src/promql/src/functions/increase.rs

```diff
@@ -0,0 +1,168 @@
+            for window in range.windows(2) {
+                let prev = window[0];
+                let curr = window[1];
+                if curr < prev {
+                    result_value += prev
+                }
+            }
+
+            result_array.push(result_value);
+        }
+
+        let result = ColumnarValue::Array(Arc::new(PrimitiveArray::from_iter(result_array)));
+        Ok(result)
+    }
+
+    pub fn scalar_udf() -> ScalarUDF {
+        ScalarUDF {
+            name: Self::name().to_string(),
+            signature: Signature::new(
+                TypeSignature::Exact(vec![Self::input_type()]),
+                Volatility::Immutable,
+            ),
+            return_type: Arc::new(|_| Ok(Arc::new(Self::return_type()))),
+            fun: Arc::new(Self::calc),
+        }
+    }
+}
+
+impl Display for Increase {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("PromQL Increase Function")
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use super::*;
+
+    fn increase_runner(input: RangeArray, expected: Vec<f64>) {
+        let input = vec![ColumnarValue::Array(Arc::new(input.into_dict()))];
+        let output = extract_array(&Increase::calc(&input).unwrap())
+            .unwrap()
+            .as_any()
+            .downcast_ref::<Float64Array>()
+            .unwrap()
+            .values()
+            .to_vec();
+        assert_eq!(output, expected);
+    }
+
+    #[test]
+    fn abnormal_input() {
+        let values_array = Arc::new(Float64Array::from_iter([
+            1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0,
+        ]));
+        let ranges = [(0, 2), (0, 5), (1, 1), (3, 3), (8, 1), (9, 0)];
+        let range_array = RangeArray::from_ranges(values_array, ranges).unwrap();
+        increase_runner(range_array, vec![1.0, 4.0, 0.0, 2.0, 0.0, 0.0]);
+    }
+
+    #[test]
+    fn normal_input() {
+        let values_array = Arc::new(Float64Array::from_iter([
+            1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0,
+        ]));
+        let ranges = [
+            (0, 2),
+            (1, 2),
+            (2, 2),
+            (3, 2),
+            (4, 2),
+            (5, 2),
+            (6, 2),
+            (7, 2),
+        ];
+        let range_array = RangeArray::from_ranges(values_array, ranges).unwrap();
+        increase_runner(range_array, vec![1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]);
+    }
+
+    #[test]
+    fn short_input() {
+        let values_array = Arc::new(Float64Array::from_iter([
+            1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0,
+        ]));
+        let ranges = [
+            (0, 1),
+            (1, 0),
+            (2, 1),
+            (3, 0),
+            (4, 3),
+            (5, 1),
+            (6, 0),
+            (7, 2),
+        ];
+        let range_array = RangeArray::from_ranges(values_array, ranges).unwrap();
+        increase_runner(range_array, vec![0.0, 0.0, 0.0, 0.0, 2.0, 0.0, 0.0, 1.0]);
+    }
+}
```

Comment:
> [Breaks in monotonicity (such as counter resets due to target restarts) are automatically adjusted for.](https://prometheus.io/docs/prometheus/latest/querying/functions/#increase)

Is this feature covered?

Reply:
nice catch. I add `counter_reset()` to cover it